modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/data/sample/MuleSampleDataService.java

```diff
@@ -0,0 +1,191 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.data.sample;
+
+import static java.lang.String.format;
+import static org.mule.runtime.api.util.Preconditions.checkArgument;
+import static org.mule.runtime.core.api.util.StringUtils.isBlank;
+import static org.mule.runtime.core.internal.util.LocationUtils.deleteLastPartFromLocation;
+import static org.mule.runtime.core.internal.util.LocationUtils.isConnection;
+import static org.mule.runtime.module.extension.internal.util.IntrospectionUtils.getImplementingName;
+import static org.mule.sdk.api.data.sample.SampleDataException.INVALID_LOCATION;
+import static org.mule.sdk.api.data.sample.SampleDataException.INVALID_TARGET_EXTENSION;
+import static org.mule.sdk.api.data.sample.SampleDataException.NOT_SUPPORTED;
+import static org.mule.sdk.api.data.sample.SampleDataException.NO_DATA_AVAILABLE;
+
+import org.mule.runtime.api.component.AbstractComponent;
+import org.mule.runtime.api.component.location.ComponentLocation;
+import org.mule.runtime.api.component.location.ConfigurationComponentLocator;
+import org.mule.runtime.api.component.location.Location;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.connector.ConnectionManager;
+import org.mule.runtime.core.api.data.sample.SampleDataService;
+import org.mule.runtime.core.api.extension.ExtensionManager;
+import org.mule.runtime.core.api.streaming.StreamingManager;
+import org.mule.runtime.core.api.util.func.CheckedSupplier;
+import org.mule.runtime.dsl.api.component.config.DefaultComponentLocation;
+import org.mule.runtime.extension.api.data.sample.ComponentSampleDataProvider;
+import org.mule.runtime.extension.api.runtime.config.ConfigurationInstance;
+import org.mule.runtime.module.extension.internal.ExtensionResolvingContext;
+import org.mule.runtime.module.extension.internal.runtime.resolver.StaticParameterValueResolver;
+import org.mule.runtime.module.extension.internal.util.ReflectionCache;
+import org.mule.sdk.api.data.sample.SampleDataException;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import javax.inject.Inject;
+
+/**
+ * Default implementation of {@link SampleDataService}
+ *
+ * @since 4.4.0
+ */
+public class MuleSampleDataService implements SampleDataService {
+
+  private ConfigurationComponentLocator componentLocator;
+  private ExtensionManager extensionManager;
+  private MuleContext muleContext;
+  private StreamingManager streamingManager;
+  private ConnectionManager connectionManager;
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Message getSampleData(Location location) throws SampleDataException {
+    boolean isConnection = isConnection(location);
```

Comment:
Should we take this into account? only operation and sources locations are valid here.